### PR TITLE
WIP: Composer support (#13)

### LINKED ISSUE #13: Making it available as composer library
Making this available as a composer library for easy maintainability.
Please let me know if that's your plan then I can help you guys migrate this to the composer library.
I'll be very happy to migrate this to the composer.

### PR DESCRIPTION
**Pull request by pushpak1300**
Resolves #13 

#### To-Do
- [X]  Add composer support
- [X] Refactor
- [X] Strict types support and PSR12 style guide  
- [x]  Update Documentation
- [ ]  Add Guzzle support
- [ ]  Testing
- [ ]  Final Refactor

### CODE REVIEW

**Comment by pushpak1300** [comment]
@vividvilla @ranjanrak 
I have drafted the PR.
I need support for the testing of some endpoint.
Can you guys help me with that if possible?

> If you have a Zerodha account then you can with prod data. Credits will be given to your account on developers.kite.trade and will be able to create the app.

can you add credit to my account if possible?

**Comment by vividvilla** [comment]
Just added some credits to your Kite connect account, you should be able to create an app and subscribe to historical API as well. Was going through the refactor and it's quite impressive actually. For testing, you can use the same mock responses as [pykiteconnect](https://github.com/zerodha/pykiteconnect/tree/master/tests/mock_responses) and it will nice if you can add some examples like here - https://github.com/zerodha/pykiteconnect/tree/master/examples

Also currently, we don't have WebSocket support on this lib but then not many actually requested as well so I am not sure if we should add that as well.

**Comment by pushpak1300** [comment]
Thanks For the Credit!
I have worked with WebSocket in PHP but I'm not sure we can add to this architecture.
Documentation is in my roadmap.

File: .github/SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Security Policy
+
+If you discover any security related issues, please email pushpak1300@gmail.com instead of using the issue tracker.
```
Comment (pushpak1300) [inline .github/SECURITY.md:3]: Please add kite zerodha email in case of security issue

File: src/KiteConnect.php [hunk 2]
```diff
@@ -0,0 +1,1773 @@
...
+ * This module provides an easy to use abstraction over the HTTP APIs.
+ * The HTTP calls have been converted to methods and their JSON responses
+ * are returned as native PHP structures, for example, dicts, lists, bools etc.
+ * See the **[Kite Connect API documentation](https://kite.trade/docs/connect/v3/)**
+ * for the complete list of APIs, supported parameters and values, and response formats.
+ * Getting started
+ * ---------------
+ * <pre>
+ * <?php
+ * include dirname(__FILE__)."/kiteconnect.php";
```
Comment (pushpak1300) [inline src/KiteConnect.php:43]: change docs

File: .github/SECURITY.md (hunk 1 above)
Comment (vividvilla) [reply .github/SECURITY.md:3]: Users can report security vulns to `bugbounty@zerodha.com `

**Comment by pushpak1300** [comment]
@vividvilla @ranjanrak I have completely refactored the code without guzzle and updated the documentation.
Please take a look at it and test it.
I have aleready tested it.
If everything works fine I will refactor to guzzle.

**Comment by pushpak1300** [comment]
Follow this guide to publish the package on composer https://laravelpackage.com/15-publishing.html#publishing-on-packagist

**Comment by pushpak1300** [comment]
@vividvilla I just tried testing with https://packagist.org/packages/pushpak1300/phpkiteconnect this package and it's working as expected.

**Comment by vividvilla** [comment]
Will go through this PR next week and plan the release.

**Comment by vividvilla** [comment]
@ranjanrak Can you please look into this if you got some time? I might not be able to do this until next weekend.

**Comment by ranjanrak** [comment]
Yeah, I'm going through this PR. Have added/modified a couple of things. Meanwhile, I felt the need to cover unit tests as well, using mock data i.e [Mockery](https://github.com/mockery/mockery). Working on that. Once completed, will raise PR for further discussion/review. @vividvilla @pushpak1300

**Comment by pushpak1300** [comment]
> Yeah, I'm going through this PR. Have added/modified a couple of things. Meanwhile, I felt the need to cover unit tests as well, using mock data i.e [Mockery](https://github.com/mockery/mockery). Working on that. Once completed, will raise PR for further discussion/review. @vividvilla @pushpak1300

Take your time let me know if anything needed from my side.

**Comment by ranjanrak** [comment]
Changes are almost done. Yesterday evening, added Guzzle HTTP client support instead of ongoing Curl. Few HTTP mocking URLs are left. Hopefully will look into this today evening and push it finally :)

**Comment by ranjanrak** [comment]
@pushpak1300 @vividvilla 
Tried a couple of times but wasn't able to push the changes here due to a permission issue. So, have raised a new PR, [refer to this](https://github.com/zerodha/phpkiteconnect/pull/15).

**Comment by pushpak1300** [comment]
Added permission for you!

**Comment by ranjanrak** [comment]
Ref [#16](https://github.com/zerodha/phpkiteconnect/pull/16)

**Comment by pushpak1300** [comment]
Thanks, @ranjanrak and @vividvilla for helping! 🚀

**Pull request closed without merging**